### ISSUE TITLE
String encoding?

### ISSUE THREAD

**Issue by bobbymcr** [JayBazuzi/Okra#6]
The project includes tchar but uses only "narrow" (ANSI) strings. Maybe explicit Unicode would be the best choice? In that case you could consider ditching the W-vs-A tchar header and use only wcout, wmain, etc. Or maybe go with utf-8, though I don't know how much complication that would add.

**Comment by JayBazuzi** [comment]
Does this have value before the project is localized to a language that wants more than ANSI?

What's the easiest way to do this and have it work on all platforms/compilers?

**Comment by bobbymcr** [comment]
I suppose it has value if you expect non-Latin characters in messages and
file names (which can and does happen even on 100% en-US systems). Perhaps
it's even more likely for locale-aware projects under test (I'm thinking of
a case where the expected and actual string values are being printed in an
assertion failure message).

Assuming that C++11 is a prereq here,
http://en.cppreference.com/w/cpp/locale/codecvt should make this relatively
straightforward and portable.

**Comment by JayBazuzi** [comment]
Yes, I definitely want people to be able to write anything they like in to the names, so fixing this makes sense.

I am disappointed that they'll be constrained to the file system name limitations, and that those rules vary by platform. They already contend with that today, and it's less constraining than identifier names, as used by many test frameworks.

**Comment by JayBazuzi** [comment]
I wrote a test for this, but I don't know enough to fix it. Maybe someone can help?

https://github.com/JayBazuzi/Okra/tree/unicode
